src/app/components/Tokens/TokenTransfers.tsx

```diff
@@ -234,16 +277,23 @@ export const TokenTransfers: FC<TokenTransfersProps> = ({
           ? [
               {
                 key: 'tokenType',
-                // TODO: temporary workaround until token type becomes available as part of RuntimeEvent
-                content: <DelayedTokenTransferTokenType event={transfer} />,
+                content: transfer.evm_log_details?.token_type ? (
+                  <TokenTypeTag tokenType={transfer.evm_log_details.token_type} />
+                ) : (
+                  <DelayedTokenTransferTokenType event={transfer} />
+                ),
                 align: TableCellAlign.Center,
               },
             ]
           : []),
         {
           key: 'value',
           align: TableCellAlign.Right,
-          content: <DelayedEventBalance event={transfer} tickerAsLink={differentTokens} />,
+          content: transfer.evm_log_details?.token_type ? (
+            <DirectEventBalance event={transfer} tickerAsLink={differentTokens} />
+          ) : (
+            <DelayedEventBalance event={transfer} tickerAsLink={differentTokens} />
```

Comment:
Names of these components are indistinguishable

Reply:
OK, so the idea was the the first one (`DirectEventBalance`) uses that that is directly available, while the second one (`DelayedEventBalance`) launches another request, and will display the data when the new server roundtrip has finished.

What names would you suggest?

Reply:
I would delete DirectEventBalance, and change EventBalance

Reply:
OK done